src/core/iomgr/pollset_posix.c

```diff
@@ -413,10 +413,10 @@ static int unary_poll_pollset_maybe_work(grpc_pollset *pollset,
 
   pfd[1].events = grpc_fd_begin_poll(fd, pollset, POLLIN, POLLOUT, &fd_watcher);
 
-  r = poll(pfd, GPR_ARRAY_SIZE(pfd), timeout);
+  r = poll(pfd, GPR_ARRAY_SIZE(pfd) - (pfd[1].events == 0), timeout);
```

Comment:
If the next commit is just a comment, I won't wait for Travis since it is currently green. I am ready to merge.